### PR TITLE
Make SUBSTRING_BYTES work similar to SUBSTRING

### PR DESCRIPTION
**Pull request by gnusi**
### Scope & Purpose

Make SUBSTRING_BYTES work similar to SUBSTRING

- [x] :hankey: Bugfix
- [ ] :pizza: New feature
- [ ] :fire: Performance improvement
- [ ] :hammer: Refactoring/simplification

### Checklist

- [x] Tests
  - [ ] **Regression tests**
  - [ ] C++ **Unit tests**
  - [x] **integration tests**
  - [ ] **resilience tests**
- [ ] :book: CHANGELOG entry made
- [ ] :books: documentation written (release notes, API changes, ...)
- [ ] Backports
  - [ ] Backport for 3.10: *(Please link PR)*
  - [ ] Backport for 3.9: *(Please link PR)*
  - [ ] Backport for 3.8: *(Please link PR)*

#### Related Information

*(Please reference tickets / specification / other PRs etc)*

- [ ] Docs PR: 
- [x] Enterprise PR: https://github.com/arangodb/enterprise/pull/1040
- [ ] GitHub issue / Jira ticket: 
- [ ] Design document:

### CODE REVIEW

File: arangod/Aql/Functions.cpp [hunk 1]
```diff
@@ -2593,24 +2593,32 @@ AqlValue functions::SubstringBytes(ExpressionContext* ctx, AstNode const& node,
...
+  int32_t const length = [&]() {
+    if (parameters.size() >= 3) {
+      return static_cast<int32_t>(
+          extractFunctionParameterValue(parameters, 2).toInt64());
+    } else {
+      return static_cast<int32_t>(str.size());
+    }
+  }();
+
+  if (length < 0 || offset >= str.size()) {
```
Comment (Dronplane) [inline arangod/Aql/Functions.cpp:2618]:
```suggestion
  if (length <= 0 || offset >= str.size()) {
```

**Review by Dronplane — APPROVED** [review]
LGTM. Minor suggestion

File: arangod/Aql/Functions.cpp (hunk 1 above)
Comment (gnusi) [reply arangod/Aql/Functions.cpp:2618]: done

File: arangod/Aql/Functions.cpp [hunk 2]
```diff
@@ -2593,24 +2593,32 @@ AqlValue functions::SubstringBytes(ExpressionContext* ctx, AstNode const& node,
...
+    }
 
-  if (str.size() < uint32_t(end)) {
+    return offset;
+  }();
+
+  int32_t const length = [&]() {
+    if (parameters.size() >= 3) {
+      return static_cast<int32_t>(
+          extractFunctionParameterValue(parameters, 2).toInt64());
```
Comment (MBkkt) [inline arangod/Aql/Functions.cpp:2612]:
```suggestion
      auto length = static_cast<int32_t>(
          extractFunctionParameterValue(parameters, 2).toInt64());
      return std::min(length, static_cast<int32_t>(str.size()));
```

File: arangod/Aql/Functions.cpp [hunk 3]
```diff
@@ -2593,24 +2593,32 @@ AqlValue functions::SubstringBytes(ExpressionContext* ctx, AstNode const& node,
     return AqlValue{AqlValueHintNull{}};
   }
 
-  int32_t const start = static_cast<int32_t>(
```
Comment (MBkkt) [inline arangod/Aql/Functions.cpp:2596]: Also probably function should be named substringBytes

**Pull request merged**